### PR TITLE
Bootguard 1.0 support

### PR DESCRIPTION
**Pull request by zaolin**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -112,18 +128,23 @@ github.com/kaey/framebuffer v0.0.0-20140402104929-7b385489a1ff/go.mod h1:tS4qtlc
...
 github.com/kr/pretty v0.1.0/go.mod h1:dAy3ld7l9f0ibDNOQOHHMYYIIbhfbHSm3C4ZsoJORNo=
+github.com/kr/pretty v0.2.1/go.mod h1:ipq/a2n7PKx3OHsz4KJII5eveXtPO4qwEXGdVfWzfnI=
 github.com/kr/pretty v0.3.0 h1:WgNl7dwNpEZ6jJ9k1snq4pZsg7DOEN8hP9Xw0Tsjwk0=
 github.com/kr/pretty v0.3.0/go.mod h1:640gp4NfQd8pI5XOwp5fnNeVWj67G7CFk/SaSQn7NBk=
 github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/pty v1.1.8/go.mod h1:O1sed60cT9XZ5uDucP5qwvh+TE3NnUj51EiZO/lmSfw=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
 github.com/kr/text v0.2.0 h1:5Nx0Ya0ZqY2ygV366QzturHI13Jq95ApcVaJBhpS+AY=
 github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
+github.com/linuxboot/fiano v5.0.0+incompatible/go.mod h1:IPKmAwYdbidivI8+nWCBO97QkdsiF8OThAHowU8Tvdk=
```
Comment (orangecms) [inline go.sum:146]: This transitively introduces a self-reference to an older version of Fiano itself via circular depenencies. Is there a chance to move the necessary bits to Fiano instead and avoid the circularity?

File: pkg/intel/metadata/bg/bgbootpolicy/bpmh.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2017-2023 the LinuxBoot Authors. All rights reserved
```
Comment (orangecms) [inline pkg/intel/metadata/bg/bgbootpolicy/bpmh.go:1]: This here is the same code as in `cbnt/cbntbootpolicy/bpmh_manifestcodegen.go` - can we get rid of that, and also other duplication?

File: pkg/intel/metadata/bg/bgkey/manifest_manifestcodegen.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+	size += s.BPKeyTotalSize()
+	size += s.KeyAndSignatureTotalSize()
+	return size
+}
+
+// PrettyString returns the content of the structure in an easy-to-read format.
+func (s *Manifest) PrettyString(depth uint, withHeader bool, opts ...pretty.Option) string {
+	var lines []string
+	if withHeader {
+		lines = append(lines, pretty.Header(depth, "Manifest", s))
```
Comment (orangecms) [inline pkg/intel/metadata/bg/bgkey/manifest_manifestcodegen.go:315]: proposal: change this into `"Key Manifest"` to make it easier to spot right away
in a similar fashion, the BPM prints as `"Boot Policy Manifest"`

Comment (orangecms) [reply pkg/intel/metadata/bg/bgkey/manifest_manifestcodegen.go:315]: https://github.com/linuxboot/fiano/pull/396

take it or leave it :)

**Review by ChriMarMe — APPROVED** [review]

**Pull request merged**